shell/src/main/java/alluxio/cli/docgen/MetricsDocGenerator.java

```diff
@@ -42,18 +44,48 @@
 @PublicApi
 public final class MetricsDocGenerator {
   private static final Logger LOG = LoggerFactory.getLogger(MetricsDocGenerator.class);
-  private static final String[] CATEGORIES =
-      new String[]{"cluster", "master", "worker", "client", "fuse"};
+  private static final String[] CATEGORIES = new String[]{
+      "client",
+      "cluster",
+      "fuse",
+      "master",
+      "worker",
+      };
   private static final String CSV_FILE_DIR = "docs/_data/table/";
   private static final String YML_FILE_DIR = "docs/_data/table/en/";
   private static final String CSV_SUFFIX = "csv";
   private static final String YML_SUFFIX = "yml";
   private static final String CSV_FILE_HEADER = "metricName,metricType";
+  private static final String TEMP_PREFIX = "temp-";
+
+  /**
+   * Compare the temp files and the committed files if validate flag is set.
+   *
+   * @param folder the location of the files to be compared
+   * @param suffix the file types (CSV/YML)
+   * @param category the category in the file names
+   * @param hasDiff a boolean indicating has there been changes
+   *
+   * @return a boolean indicating if there is anything difference
+   * between the temp files and committed files
+   */
+  public static boolean compareFiles(String folder, String suffix, String category,
+                                     boolean hasDiff) throws IOException {
+    if (!FileUtils.contentEquals(new File(folder, (TEMP_PREFIX + category + "-metrics." + suffix)),
+        new File(folder, (category + "-metrics." + suffix)))) {
+      hasDiff = true;
+      System.out.println("Metrics file " + category + "-metrics." + suffix + " changed.");
+    }
+    File tempFile = new File(folder, TEMP_PREFIX + category + "-metrics." + suffix);
+    tempFile.delete();
```

Comment:
compareFiles shouldn't be responsible for deleting the temp file...

okay i see that the logic that calls the metrics version of compareFiles doesn't have the same try/finally block as in the configuration version. the try/finally is actually critical to proper control flow. if compareFile throws an exception, such as in the case where one of the files doesn't exist and `FileUtils.contentEquals` throws an IOException, it will leave all the temporary files undeleted